crates/bevy_ecs/src/schedule/set.rs

```diff
@@ -202,14 +310,197 @@ mod tests {
         schedule.add_systems(|mut flag: ResMut<Flag>| flag.0 = true);
         world.add_schedule(schedule, A);
 
-        let boxed: Box<dyn ScheduleLabel> = Box::new(A);
+        let interned = A.intern();
 
         world.insert_resource(Flag(false));
-        world.run_schedule(&boxed);
+        world.run_schedule(interned);
         assert!(world.resource::<Flag>().0);
 
         world.insert_resource(Flag(false));
-        world.run_schedule(boxed);
+        world.run_schedule(interned);
         assert!(world.resource::<Flag>().0);
     }
+
+    #[test]
+    fn test_derive_schedule_label() {
```

Comment:
Can you add test cases for non-unit fieldless structs and enums?

```rust
struct WeirdUnit1 { }

struct WeirdUnit2();

enum WeirdUnitEnum {
   Variant1 {},
    Variant2()
}
```

Reply:
I've added test cases for empty struct and tuple variants, both standalone and as part of an enum. I had to adjust `static_ref_impl` for it to work and stumbled across some weird behavior that is only present in the current stable channel, preventing me from using `&Self::EmptyTuple()` and having to use `&EnumLabel::EmptyTuple()` instead.  

On the Rust playground, [this](https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=c544fa6b762881a53a46c8aae4e702a6) compiles, but [this](https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=39468824e70ec79869792d83900d95f6) doesn't. Both versions work on the current beta channel.

Reply:
You should be able to just do `&Self {}`, which would work for unit structs, as well as tuple and named structs with no fields.

Reply:
Ah, you're right. That's very useful to know.